### PR TITLE
Adds plain condition for itemtypes

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
- Adds plain condition for itemtypes.
- If an ItemData is created using Material it **should** be plain, don't know why everything is set to false for plain unless used by the expression.

---
**Target Minecraft Versions:* none
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/5616

### CODE REVIEW

File: src/test/skript/tests/syntaxes/conditions/CondIsPlain.sk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+test "itemtype is plain":
```
Comment (APickledWalrus) [inline src/test/skript/tests/syntaxes/conditions/CondIsPlain.sk:1]: should include testing for items that aren't plain

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
Looks pretty good

Are you sure setting the plain flag in the constructor is safe? I likely did it because `item is a plain diamond` and `item is a diamond` have different functionality (and the latter should not have the plain flag set to true)

in fact now that I think about it that's probably why there wasn't a condition in the first place 🤣 
this condition may *really* need to also compare the input ItemType to a new ItemType of the same material with the plain flag set to true

**Comment by sovdeeth** [comment]
This also does not work for things like `if player's tool is plain`.
image
image

File: src/test/skript/tests/syntaxes/conditions/CondIsPlain.sk (hunk 1 above)
Comment (TheLimeGlass) [reply src/test/skript/tests/syntaxes/conditions/CondIsPlain.sk:1]: Every item not done with the plain expression are not plain which is why I found it dumb that a simple `emerald` call was considered not plain some how.

**Comment by TheLimeGlass** [comment]
> This also does not work for things like `if player's tool is plain`. image image

All items are not plain unless used with the plain expression.

This is why i'm pushing to implement Material made ItemData as plain. There are other calls of Skript that don't use this constructor for some reason. Majority is used through ItemData(ItemStack)

**Comment by sovdeeth** [comment]
> All items are not plain unless used with the plain expression.
> 
> This is why i'm pushing to implement Material made ItemData as plain. There are other calls of Skript that don't use this constructor for some reason. Majority is used through ItemData(ItemStack)

Ah sounds good. I don't think this PR should be added until that's implemented, though, because right now the behavior is rather unintuitive.

**Review by Moderocky — COMMENTED** [review]
I'm not entirely sure what the current objective for this syntax is. It seems like the only things that are plain are the things declared as `plain`. If you have to declare it as such yourself then what does the condition help with?

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
I still have the same concerns from my previous comment:

Are you sure setting the plain flag in the constructor is safe? I likely did it because `item is a plain diamond` and `item is a diamond` have different functionality (and the latter should not have the plain flag set to true)

this condition may really need to compare the input ItemType to a new ItemType of the same material with the plain flag set to true

**Comment by sovdeeth** [comment]
Closing due to inactivity

**Pull request closed without merging**